src/javascript/app/Stores/Modules/SmartChart/smart-chart-store.js

```diff
@@ -40,6 +40,16 @@ export default class SmartChartStore extends BaseStore {
     @observable trade_chart_layout   = null;
     trade_chart_symbol               = null;
 
+    @action.bound
+    switchToContractMode() {
+        this.saveAndClearTradeChartLayout('contract');
+        this.setContractMode(true);
+        // In contract mode, the chart granularity should always be 0
+        // and the chart_type should always be 'mountain'
```

Comment:
Isn't there candle contracts in contract mode also?

Reply:
ohh ya, it should show candle charts for contracts > 1 hour

Reply:
@oskar it is handled in contract_store